src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDashboard/requestLog.tsx

```diff
@@ -0,0 +1,331 @@
+  let priority = 'error';
+  if (code <= 399 && code >= 300) {
+    priority = 'warning';
+  } else if (code <= 299 && code >= 100) {
+    priority = 'success';
+  }
+
+  return (
+    <div>
+      <Tag priority={priority}>{code === 0 ? 'timeout' : code}</Tag>
+    </div>
+  );
+};
+
+const TimestampLink = ({date, link}: {date: moment.MomentInput; link?: string}) => {
+  return link ? (
+    <Link to={link}>
+      <DateTime date={date} />
+    </Link>
+  ) : (
+    <DateTime date={date} />
+  );
+};
+
+type Props = AsyncComponent['props'] & {
+  app: SentryApp;
+};
+
+type State = AsyncComponent['state'] & {
+  eventType: string;
+  errorsOnly: boolean;
+  currentPage: number;
+  requests: SentryAppWebhookRequest[];
+};
+
+export default class RequestLog extends AsyncComponent<Props, State> {
+  shouldReload = true;
+
+  get eventTypes() {
+    // TODO(nola): ideally this would be kept in sync with EXTENDED_VALID_EVENTS on the backend
+
+    const {app} = this.props;
+    let events = [ALL_EVENTS, 'installation.created', 'installation.deleted'];
+
+    if (app.events.includes('error')) {
+      events = [...events, 'error.created'];
+    }
+    if (app.events.includes('issue')) {
+      events = [
+        ...events,
+        'issue.created',
+        'issue.resolved',
+        'issue.ignored',
+        'issue.assigned',
+      ];
+    }
+    if (app.isAlertable) {
+      events = [...events, 'event_alert.triggered'];
+    }
+
+    const issueLinkComponent: any = (app.schema.elements || []).find(
+      (element: any) => element.type === 'issue-link'
+    );
+    if (issueLinkComponent) {
+      const issueLinkEvents = ['external_issue.created', 'external_issue.linked'];
+      if (
+        (issueLinkComponent.create && issueLinkComponent.create.uri) ||
+        (issueLinkComponent.link && issueLinkComponent.link.uri)
+      ) {
+        issueLinkEvents.push('select_options.requested');
+      }
+      events = [...events, ...issueLinkEvents];
+    }
+
+    return events;
+  }
+
+  get hasNextPage() {
+    return (this.state.currentPage + 1) * MAX_PER_PAGE < this.state.requests.length;
+  }
+
+  get hasPrevPage() {
+    return this.state.currentPage > 0;
+  }
+
+  getEndpoints(): Array<[string, string, any] | [string, string]> {
+    const {slug} = this.props.app;
+
+    const query: any = {};
+    if (this.state) {
+      if (this.state.eventType !== ALL_EVENTS) {
+        query.eventType = this.state.eventType;
+      }
+      if (this.state.errorsOnly) {
+        query.errorsOnly = true;
+      }
+    }
+
+    return [['requests', `/sentry-apps/${slug}/requests/`, {query}]];
+  }
+
+  getDefaultState() {
+    return {
+      ...super.getDefaultState(),
+      requests: [],
+      eventType: ALL_EVENTS,
+      errorsOnly: false,
+      currentPage: 0,
+    };
+  }
+
+  onChangeEventType = eventType => {
+    this.setState({
+      eventType,
+      currentPage: 0,
+    });
+    this.remountComponent();
+  };
+
+  onChangeErrorsOnly = () => {
```

Comment:
Can we rename this to be `handle*`? https://github.com/getsentry/frontend-handbook#event-handlers